### DISCUSSION TITLE
reconfigure the storage location

### DISCUSSION THREAD

**Discussion by macomar45** [ciur/papermerge#475] [Q&A]
Hello,
we have installed PM via Docker on Debian 11 (vSphere 7 and SAN Storage). Now we want to reconfigure the storage location to another larger partition for the amount of data we want to archive. Can we be helped here how this is done in Docker ?
Thanks a lot.

**Comment by w4tzmann** [comment]
Hello,
can you share your docker-compose.yml? It should be easier that way to help with configuration. Of course, you can replace all sensitive data.

In the example docker-compose.yml it looks like this:

> volumes:
>       - media_root:/opt/media


So, the Docker volume "media_root" needs to be moved. There should be instructions for this on the internet. If a mounted folder is used, you can simply move the folder to the new location and adjust the path, e.g:

> volumes:
>       - /opt/papermerge_data:/opt/media

**Reply by macomar45** [reply]
Hello,
thanks for your answer but I'm a bit lost as to how to implement it.

We have created a native partition "Data" on the Linux system where the Docker is running and we want the data from Papermerge to be written to the mounted partition "Data".

ACTUAL:
/var/lib/docker/volumes/docker_media_root/_data

TARGET:
/data

Here we don't know how this should be done. All attempts ended up in a crashed system.

**Reply by w4tzmann** [reply]
Hey,

I'm far from an specialist in this targes. Currently I use bind mounts almost exclusively.
For papermerge I replaced 

> volumes:
> - media_root:/opt/media

with
> volumes:
> - /data/papermerge/data:/opt/media

If you have already saved some documents, you will need to copy/ move this files from the exisitng volume. (Something like "cp /var/lib/docker/volumes/docker_media_root/_data/* /data/papermerge/data" should work if you stop the container before.)

The simplest way to move all docker data (but this also affects e.g. the images), wil be to change the "data-root" in the docker daemon.json

> 1. stop Docker
> 2. move the files, for example: sudo mv /var/lib/docker /data/docker
> 3. change the settings in  /etc/docker/daemon.json to something like this:
> {
> "data-root": "/data/docker"
> }
> 4. start docker again
> 5. 5. check with docker info if the new path is recognized
> 6. Restart the containers (docker-compose up -d)
> 
> [Source](https://www.bananatronics.org/how-to-move-docker-volumes/#:~:text=First%20stop%20the%20docker%20service,docker%20to%20the%20new%20location.&text=Next%20the%20configuration%20of%20the,new%20location%20of%20the%20volumes.&text=The%20next%20step%20may%20not,habit%20to%20do%20it%20anyway.)

Also a symlink could work.
